File: 8-String-to-Integer-(atoi).cpp
class Solution {
public:
    int myAtoi(string s) {
        bool spacesIgnored=false, signPassed=false;
        int n=0, sign=1;
        vector <int> nums;

        for (char c : s){
            if ('0'<=c && c<='9'){
                spacesIgnored=signPassed=true;
                nums.push_back(c-'0');
            }
            else if (c=='-' && !signPassed) {
                spacesIgnored= signPassed= true;
                sign = -1;
            }
            else if (c=='+' && !signPassed) {
                spacesIgnored= signPassed= true;
            }
            else if (c==' ' && !spacesIgnored) continue;
            else break;
        }

        for (int i=nums.size()-1; i>=0; i--)
        {
            if (n+nums[i]*pow(10,nums.size()-1-i)>INT_MAX){
                n = sign==1 ? INT_MAX : INT_MIN;
                return n;
            }
            else n += nums[i]*pow(10,nums.size()-1-i);
        }
        
        return n*sign;
    }
};

File: 8-String-to-Integer-(atoi).java
class Solution {
    public int myAtoi(String s) {
        byte sign=1;
        int number=0, n=s.length();
        int i=0;
        

        while(i<n && s.charAt(i)==' ') {
            i++;
        }
        if (i>=n) return 0;
        
        if (s.charAt(i)=='-') {
            sign = -1;
            i++;
        }
        else if (s.charAt(i)=='+') i++;
        else if (!(s.charAt(i)>='0' && s.charAt(i)<='9')) return number;
        
        while(i<n){
            if(s.charAt(i)>='0' && s.charAt(i)<='9'){
                if (10*(long)number + (s.charAt(i)-'0') > Integer.MAX_VALUE){
                    return sign == 1 ? Integer.MAX_VALUE : Integer.MIN_VALUE;
                }
                number = 10*number + (s.charAt(i)-'0');
                i++;
            }
            else return number*sign;
        }
        return number*sign;
    }
}